relay-server/src/utils/metrics_rate_limits.rs

```diff
@@ -73,6 +71,56 @@ pub fn extract_transaction_count(
     Some(TransactionCount { count, has_profile })
 }
 
+/// Extracts quota information from a list of metric buckets.
+pub fn extract_metric_quantities<'a, I, V>(buckets: I, mode: ExtractionMode) -> SourceQuantities
+where
+    I: IntoIterator<Item = V>,
+    BucketView<'a>: From<V>,
+{
+    let mut quantities = SourceQuantities::default();
+
+    for bucket in buckets {
+        quantities.buckets += 1;
+        if let Some(count) = count_metric_bucket(bucket.into(), mode) {
+            quantities.transactions += count.count;
+            if count.has_profile {
+                quantities.profiles += count.count;
+            }
+        }
+    }
+
+    quantities
+}
+
+pub fn reject_metrics(
+    addr: Addr<TrackOutcome>,
+    quantities: SourceQuantities,
```

Comment:
Could take the `IntoIterator<Item = impl From<BucketView<'a>>>` directly here

Reply:
That would be a bit less practical, as computing source quantities also needs the extraction mode from the project state. This signature should encourage reusing the quantities as much as possible, since those are expensive to create.